pkg/registry/populator.go

```diff
@@ -407,3 +411,54 @@ func DecodeFile(path string, into interface{}) error {
 
 	return decoder.Decode(into)
 }
+
+// ValidateEdgeBundlePackage ensures that all bundles in the input will only skip or replace bundles in the same package.
+func (i *DirectoryPopulator) ValidateEdgeBundlePackage(images []*ImageInput) error {
+	// track packages for encountered bundles
+	expectedBundlePackages := map[string]string{}
+	for _, b := range images {
+		r, err := b.Bundle.Replaces()
+		if err != nil {
+			return fmt.Errorf("failed to validate replaces for bundle %s(%s): %v", b.Bundle.Name, b.Bundle.BundleImage, err)
+		}
+
+		skipped, err := b.Bundle.Skips()
+		if err != nil {
+			return fmt.Errorf("failed to validate skipped entries for bundle %s(%s): %v", b.Bundle.Name, b.Bundle.BundleImage, err)
+		}
+
+		for _, bndl := range append(skipped, r, b.Bundle.Name) {
+			if len(bndl) == 0 {
+				continue
+			}
+
+			if pkg, ok := expectedBundlePackages[bndl]; ok && pkg != b.Bundle.Package {
+				pkgs := []string{pkg, b.Bundle.Package}
+				sort.Strings(pkgs)
+				return fmt.Errorf("bundle %s must belong to exactly one package, found on: %v", bndl, pkgs)
+			}
+			expectedBundlePackages[bndl] = b.Bundle.Package
+		}
+	}
+	if len(expectedBundlePackages) == 0 {
+		return nil
+	}
+
+	pkgs, err := i.querier.ListPackages(context.TODO())
```

Comment:
Ideally, we'd place this kind of check in the table schema itself(eg put a constraint in the replaces/skips fields in the operatorbundle table to allow bundle names that belong to the same package only). Calling the querier, i.e opening a db connection every single time is an expensive operation. Opening a db connection for every additional step we take while loading a bundle = making the load operation a little bit more inefficient every time. 

Having said that, if this is being considered as a quick dirty fix for the time sqllite operations are deprecated, and we're okay with the load operation being a little bit inefficient, consider this comment withdrawn.

Reply:
We could add it to schema, but we'd also need to enforce the one package per bundle rule or this still happens. It's not just making sure it's in the right package, but that it isn't in any other package.

Reply:
Shouldn't we be doing that anyway then?

Reply:
We can have a trigger checking for this but with the way adding bundles in sqlite works, that alone would get us bundles that are partially added (i.e, in `operatorbundle` table alone with some entries into `channel_entry` table). Considering we also nuke the package for reconstructing bundle graphs, having that start and fail half way would be really bad.

So we'd need to do this check before starting the add anyway, since our bundle adds aren't one atomic operation.

Reply:
I don't think we should touch the schema or make a db migration unless we absolutely have to

Reply:
That's a good point too.